app/views/admin/contacts/index.html.erb

```diff
@@ -1,7 +1,7 @@
 <% page_title @contactable.name %>
 <div class="organisation-header">
   <h1><%= @contactable.name %></h1>
-  <%= link_to "View on website", @contactable %>
+  <%= link_to "View on website", show_url_with_public_and_cachebusted_options(@contactable) %>
```

Comment:
I like this generic URL approach. I think we could go further and create a `view_on_website_link_for(model)` helper that generates the entire `link_to` (I'm not wedded to that naming).